src/examples/simple_snapp.ts

```diff
@@ -20,8 +20,8 @@ class SimpleSnapp extends SmartContract {
     this.x.set(x);
   }
 
-  @method async update(y: Field) {
-    let x = await this.x.get();
+  @method update(y: Field) {
+    let x = this.x.get();
```

Comment:
I laid out here how I think it will work: https://github.com/o1-labs/snarkyjs/issues/113

The only alternative I see would be to refactor Pickles so that it can handle async circuits. This would be better for users (enables them to make their methods async, which adds a lot of flexibility), but is probably much harder to implement.